pkg/kubelet/nodelease/controller_test.go

```diff
@@ -282,3 +282,111 @@ func TestRetryUpdateLease(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateUsingLatestLease(t *testing.T) {
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			UID:  types.UID("foo-uid"),
+		},
+	}
+
+	gr := schema.GroupResource{Group: "v1", Resource: "lease"}
+	notFoundErr := apierrors.NewNotFound(gr, "foo")
+	cases := []struct {
+		desc                       string
+		latestLease                *coordinationv1.Lease
+		updateReactor              func(action clienttesting.Action) (bool, runtime.Object, error)
+		getReactor                 func(action clienttesting.Action) (bool, runtime.Object, error)
+		createReactor              func(action clienttesting.Action) (bool, runtime.Object, error)
+		onRepeatedHeartbeatFailure func()
+		expectLeaseResourceVersion string
+	}{
+		{
+			desc:          "latestLease is nil and need to create",
+			latestLease:   nil,
+			updateReactor: nil,
+			getReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, nil, notFoundErr
+			},
+			createReactor: func(action clienttesting.Action) (b bool, object runtime.Object, e error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "1",
+		},
+		{
+			desc:        "latestLease is nil and need to update",
+			latestLease: nil,
+			updateReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "2"}}, nil
+			},
+			getReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "2",
+		},
+		{
+			desc:        "latestLease exist and need to update",
+			latestLease: &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}},
+			updateReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "2"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "2",
+		},
+		{
+			desc:        "update with latest lease failed",
+			latestLease: &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}},
+			updateReactor: func() func(action clienttesting.Action) (b bool, object runtime.Object, e error) {
```

Comment:
remove names of output params, i.e.

func (action clienttesting.Action) (bool, runtime.Object, error) {